src/boot/measure.c

```diff
@@ -517,6 +518,27 @@ static int measure_kernel(PcrState *pcr_states, size_t n) {
                         m += sz;
                 }
 
+                if (c == UNIFIED_SECTION_LINUX) {
+                        _cleanup_free_ PeHeader *pe_header = NULL;
+
+                        r = pe_load_headers(fd, /*ret_dos_header=*/ NULL, &pe_header);
+                        if (r < 0)
+                                log_warning_errno(r, "Failed to parse kernel image file '%s', ignoring: %m", arg_sections[c]);
+                        else {
+                                memzero(buffer, BUFFER_SIZE);
+
```

Comment:
please add a comment explaining what goes on here, i.e. that we measure a bunch of NUL bytes here, to make up for the difference between SizeofOfImage and VirtualSize